storage/filesystem/dotgit/dotgit.go

```diff
@@ -655,24 +655,47 @@ func (d *DotGit) ObjectDelete(h plumbing.Hash) error {
 	return err1
 }
 
+type refError string
+
+func (r refError) Error() string {
+	return string(r)
+}
+
+var emptyRefFileError = refError("The file is empty")
```

Comment:
Please move this to the top together with the other errors.

```suggestion
var ErrEmptyRefFile = errors.New("ref file is empty")
```